### PR TITLE
Treat the 'algorithm' parameter in Digest HTTP authentication as case-insensitive.

### PR DESCRIPTION
**Pull request by filipnavara**
Issue #32408

The HTTP Digest authentication is specified in [RFC 7616](https://tools.ietf.org/html/rfc7616.html). It specifies the syntax using [ABNF](https://tools.ietf.org/html/rfc5234), which explicitly states `ABNF strings are case insensitive and the character set for these strings is US-ASCII.`

This solves a compatibility issue when connecting to certain servers that send "md5" in lower case. This previously worked with the WinHTTP handler and the bug only affects the new Sockets handler. A backport to 2.1 and 2.2 should be considered.

### CODE REVIEW

File: src/Common/tests/System/Net/Http/LoopbackServer.AuthenticationHelpers.cs [hunk 1]
```diff
@@ -248,11 +248,11 @@ internal static bool IsDigestAuthTokenValid(string clientResponse, string reques
             }
 
             if (string.IsNullOrEmpty(algorithm))
-                algorithm = "sha-256";
+                algorithm = "MD5";
```
Comment (filipnavara) [inline src/Common/tests/System/Net/Http/LoopbackServer.AuthenticationHelpers.cs]: This updates the loopback server to use the correct default algorithm as specified by RFC 7616: `If this is not present, it is assumed to be "MD5".` The client-side code already used the correct default.

**Comment by davidsh** [comment]
@dotnet-bot test Outerloop Windows x64 Debug Build
@dotnet-bot test Outerloop Linux x64 Debug Build
@dotnet-bot test Outerloop NETFX x86 Debug Build
@dotnet-bot test Outerloop UWP CoreCLR x64 Debug Build
@dotnet-bot test Outerloop OSX x64 Debug Build

**Comment by davidsh** [comment]
CI failures are unrelated to PR.

**Comment by filipnavara** [comment]
Most of the outer loop failures seem unrelated, but there is couple of them related to authentication. I will be back on Monday, so if no one looks into it earlier I will do it by then.

**Comment by filipnavara** [comment]
Thanks!

**Pull request merged**